packages/client/src/base/actions/focus-change-action.ts

```diff
@@ -13,14 +13,34 @@
  *
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
  ********************************************************************************/
-import { Action } from '@eclipse-glsp/protocol';
+import { Action, hasBooleanProp } from '@eclipse-glsp/protocol';
 
-export class FocusStateChangedAction implements Action {
-    static KIND = 'focusStateChanged';
-    readonly kind = FocusStateChangedAction.KIND;
-    constructor(public readonly hasFocus: boolean) {}
+/**
+ * A `FocusStateChangedAction` is dispatched by the client whenever the
+ * diagram focus changes (i.e. focus loss or focus gain).
+ */
+export interface FocusStateChangedAction extends Action {
+    /**
+     * The unique action kind
+     */
```

Comment:
I'd spare such comments, also if it hurts to be "incomplete". But imho they just add noise and add no information.